CONTRIBUTING.md

```diff
@@ -1,143 +1,177 @@
+If something about the setup is unclear, or you run into an error, there are [discussions](https://github.com/ecamp/ecamp3/discussions) on GitHub for you to ask questions and ask for help. :computer:
+### Labels :label:
+Issues are marked with labels and some of them are not self-explanatory and are explained here:
+- **Type-Labels**:
+  Type labels follow the `type: *` format with the options `Frontend`, `Print`, `Deployment` & `API` the architecture for those are partially documented in the [wiki](https://github.com/ecamp/ecamp3/wiki/architecture-frontend)
+- **Needs prototype**: :bulb: If you have an idea how to solve this issue: we'd like to see it. This issue needs a prototype before actual implementation begins since the specifications are somewhat vague. A prototype can be many things, whether your prototype is a sketch, mockup, partial implementation or something else is up to you.
+- **Good first issue**: :green_heart: Beginner friendly issues.
+- **Feature request**: :rocket: An idea/request for a functionality, not ready to be implemented but ready to be discussed.
 
-### Git setup
+### :point_right: Starting with an issue 
+To get started, find an issue that interests you. If you are new, we recommend selecting a [Good first issue](https://github.com/ecamp/ecamp3/labels/Good%20first%20issue).
+Please note that for other issues we recommend ones with the label [Ready for Implementation](https://github.com/ecamp/ecamp3/issues?q=is%3Aopen+is%3Aissue+label%3A%22Ready+for+implementation%22), which signifies that the issue should have clear definitions. If you have any questions, feel free to ask.
+If you are working on an issue, please leave a comment so that we can assign it to you, to make sure that the specifications are still up-to-date, and to prevent two people working on the same issue.
+Alternatively, open a draft pull request and mention the issue ID to signal that you are working on that particular issue.
+Please note that while the wiki can be helpful in understanding the project, it's not exhaustive (meaning there might be parts missing or out of date). If you have any questions, please comment on the issue for clarification. We are happy to help and answer any questions you might have.
+
+### Git setup :octocat:
 
 We use a triangular git workflow. This means that all changes are first pushed to a contributor's fork of the repository, and then the changes are merged into the main fork via a pull request. In practice, setting up this workflow looks as follows:
 
-1. Fork the main repository onto your GitHub account. Let's say your GitHub account is called `your-username`.
+1. Fork the main repository onto your GitHub account. To use the commands your configured git `user.name` must be exactly your git user name. 
```

Comment:
> To use the commands your configured git `user.name` must be exactly your git user name.
Is this a requirement by us?

Github just takes the user.name used in the commit.

Reply:
This is one of my bolder choices in this PR. Here is what I was thinking: the people who lack the technical knowledge to set up a git repo using a forked repo are the main concerns here. When this was the case for me I used Git over GUI Tools (Login to GitHub in Intellij, GitHub Desktop) and my Git username was always set to my GitHub username. I made the assumption that most people have `git username = GitHub username` (I call such cases 'happy flow', not sure if it's a common term).

In the happy flow, you can now execute every command as it is (for example using IntelliJ or copy-paste-execute) and there are no modifications needed.

If you use a different git username the only thing that changed compared to the previous version is the text you are replacing. You still need to replace text in the exact same places. Instead of replacing `your-username` (I believe that was the previous placeholder) you need to replace `$(git config user.name)`. This is a bit more confusing than before, I would agree. I made the assumption that people who need to replace this and don't fall into the happy flow are a minority with the technical understanding to set this up without documentation.

There is no requirement to have your git username match your GitHub username, this is a requirement for the happy flow. This could probably be better written.

This might be an impactful change but I am convinced it will make the setup more efficient overall. **Do you want me to clarify in the documentation or revert my changes?**

Reply:
I like the idea of describing the happy flow for beginners, and leave experienced git gurus to do things their way. Maybe you could just change "must" (and "need to" in the paragraph below) to "should", and it'll sound less harsh while still being strongly suggestive?